ert/serialization/_serializer.py

```diff
@@ -45,8 +69,47 @@ def encode(self, obj: Any, *args: Any, **kwargs: Any) -> str:
     def decode(self, series: str, *args: Any, **kwargs: Any) -> Any:
         return yaml.safe_load(series)
 
-    def encode_to_file(self, obj: Any, fp: TextIO, *args: Any, **kwargs: Any) -> None:
-        yaml.dump(obj, fp)
+    def encode_to_file(
+        self, obj: Any, filehandle: TextIO, *args: Any, **kwargs: Any
+    ) -> None:
+        yaml.dump(obj, filehandle)
+
+    def decode_from_filehandle(
+        self, filehandle: TextIO, *args: Any, **kwargs: Any
+    ) -> Any:
+        return yaml.safe_load(filehandle)
+
+    async def decode_from_path(
+        self, path: Union[str, Path], *args: Any, **kwargs: Any
+    ) -> Any:
+        async with aiofiles.open(path, mode="rt", encoding="utf-8") as filehandle:
+            contents = await filehandle.read()
+            return self.decode(contents, *args, **kwargs)
+
+
+class _ecl_sum_serializer(Serializer):
+    def encode(self, obj: Any, *args: Any, **kwargs: Any) -> str:
+        raise NotImplementedError("not implemented")
+
+    def decode(self, series: str, *args: Any, **kwargs: Any) -> Any:
+        raise NotImplementedError("not implemented")
+
+    def encode_to_file(
+        self, obj: Any, filehandle: TextIO, *args: Any, **kwargs: Any
+    ) -> None:
+        raise NotImplementedError("not implemented")
+
+    def decode_from_filehandle(
+        self, filehandle: TextIO, *args: Any, **kwargs: Any
+    ) -> Any:
+        raise NotImplementedError("not implemented")
+
+    async def decode_from_path(
```

Comment:
How does this "fix" having to deal with UNSMRY+SMSPEC?

Reply:
The `EclSum` object deals with this, we just need to provide the Eclipse base-name as a path with a random suffix after the last `.`, and then it will compute the actual paths it needs.